user-interface/src/main/java/life/qbic/datamanager/views/notifications/Toast.java

```diff
@@ -0,0 +1,248 @@
+package life.qbic.datamanager.views.notifications;
+
+import static java.util.Objects.nonNull;
+import static java.util.Objects.requireNonNull;
+
+import com.vaadin.flow.component.AttachEvent;
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.ComponentEventListener;
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.button.Button;
+import com.vaadin.flow.component.button.ButtonVariant;
+import com.vaadin.flow.component.html.Div;
+import com.vaadin.flow.component.notification.Notification;
+import com.vaadin.flow.router.BeforeLeaveListener;
+import com.vaadin.flow.router.RouteParameters;
+import com.vaadin.flow.router.RouterLink;
+import com.vaadin.flow.shared.Registration;
+import com.vaadin.flow.theme.lumo.LumoIcon;
+import java.util.ArrayList;
+import java.util.List;
+import life.qbic.datamanager.views.general.ComponentFunctions;
+
+/**
+ * A toast notification is a non-modal, unobtrusive window element. It is meant to show information
+ * in brief and auto-expiring windows to the user.
+ * <p>
+ * It is not meant to require user interaction but may provide optional ways for the user to
+ * interact with it.
+ *
+ * @since 1.4.0
+ */
+public class Toast extends Notification {
+
+  protected Component content;
+  private boolean closeable;
+  private final List<Registration> closeOnNavigationListeners = new ArrayList<>();
+  private Type type;
+  private final Button closeButton;
+
+  protected enum Type {
+    SUCCESS,
+    INFO;
+  }
+
+  protected Toast() {
+    super();
+    addClassName("toast-notification");
+    closeButton = new Button(LumoIcon.CROSS.create());
+    closeButton.addThemeVariants(ButtonVariant.LUMO_TERTIARY_INLINE);
+    closeButton.addClickListener(it -> this.close());
+    closeButton.addClassName("close-button");
+    closeButton.setVisible(false);
+    add(closeButton);
+    setCloseable(true);
+    setPosition(Position.BOTTOM_START);
+    setType(Type.INFO);
+    closeOnNavigation(true);
+    setDuration(5_000);
+  }
+
+  /**
+   * Sets whether toasts are closed after navigation. By default, Toasts no not stay open after
+   * navigation.
+   *
+   * @param close whether to close the toast notification on navigation or not. true - closes the
+   *              toast after navigation.
+   * @return the modified toast
+   */
+  public Toast closeOnNavigation(boolean close) {
```

Comment:
Why not using the builder pattern here, to create an immutable Toast object? It can still be configured upon creation, but I dont understand the use case of having its configuration mutable. I have more concern about opening the door fot side effects in applications.

Can you elaborate some strong arguments for mutability, and if it is needed for ALL behaviour related properties? If not, i suggest to make them mutable-